zbx_nginx_stats.py

```diff
@@ -1,31 +1,50 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
-import urllib2, base64, re, struct, time, socket, sys, datetime, os.path
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import sys
+is_python_3 = sys.version_info > (3, 0)
+
+import argparse, base64, re, struct, time, socket, datetime, os.path
+
+if is_python_3:
+    import urllib.request as urllib # Python 3
+    import codecs
+else:
+    import urllib2 as urllib # Python 2
 
 try:
     import json
 except:
     import simplejson as json
 
-zabbix_host = '127.0.0.1'   # Zabbix server IP
-zabbix_port = 10051         # Zabbix server port
-hostname = 'Zabbix Agent'   # Name of monitored host, like it shows in zabbix web ui
-time_delta = 1              # grep interval in minutes
-
-# URL to nginx stat (http_stub_status_module)
-stat_url = 'https://nginx.server/nginx_stat'
-
-# Nginx log file path
-nginx_log_file_path = '/var/log/nginx/access.log'
-
-# Optional Basic Auth
-username = 'user'
-password = 'pass'
+parser = argparse.ArgumentParser()
+parser.add_argument('--use-ssl', default=False, action="store_true", help='Enable SSL')
+parser.add_argument('--ssl-cafile', default='/etc/zabbix/certs/cacert.pem', type=str, help='CA PEM file path')
+parser.add_argument('--ssl-certfile', default='/etc/zabbix/certs/agent.crt', type=str, help='Client/Agent certificate file path')
+parser.add_argument('--ssl-keyfile', default='/etc/zabbix/certs/agent.key', type=str, help='Client/Agent private key file path')
+parser.add_argument('--zabbix-server', default='127.0.0.1', type=str, help='Zabbix server host')
+parser.add_argument('--zabbix-port', default=10051, type=int, help='Zabbix server port')
+parser.add_argument('--monitored-hostname', default='Zabbix agent', type=str, help='Name of monitored host, like it shows in zabbix web ui')
+parser.add_argument('--nginx-auth-username', default=None, type=str, help='Nginx authentication username')
+parser.add_argument('--nginx-auth-password', default=None, type=str, help='Nginx authentication password')
+parser.add_argument('--nginx-status-module-url', default='http://127.0.0.1:55777/nginx-status', type=str, help='URL to retrieve Nginx status (http_stub_status_module)')
```

Comment:
Sorry, it was something specific of our nginx configuration as we don't expose the stats module to the world. I should have changed it back to the default HTTP port.